clustered/server/src/main/java/org/ehcache/clustered/server/store/MessageToTrackerSegmentFunction.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.server.store;
+
+import org.ehcache.clustered.common.internal.messages.EhcacheEntityMessage;
+import org.terracotta.entity.ConcurrencyStrategy;
+
+import java.util.function.ToIntFunction;
+
+/**
+ * MessageToTrackerSegmentFunction
+ */
+class MessageToTrackerSegmentFunction implements ToIntFunction<EhcacheEntityMessage> {
+
+  private ConcurrencyStrategy<EhcacheEntityMessage> concurrencyStrategy;
+
+  MessageToTrackerSegmentFunction(ConcurrencyStrategy<EhcacheEntityMessage> concurrencyStrategy) {
+
+    this.concurrencyStrategy = concurrencyStrategy;
+  }
+  @Override
+  public int applyAsInt(EhcacheEntityMessage value) {
+    // Concurrency is 1 based, segments are 0 based
+    return concurrencyStrategy.concurrencyKey(value) - 1;
```

Comment:
Can probably be just a lambda `message -> concurrencyStrategy.concurrencyKey(message) - 1`. But that's just a matter of taste

Reply:
But that is a magic number in its own right and I did not want to duplicate it between active and passive

Reply:
Right. Ok